lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -411,4 +430,56 @@ sub _processMSIE {
 
 }
 
+# List of SQL Instances
+sub _getSqlInstancesList {
+    my (%params) = @_;
+    my $logger = $params{logger};
+	my $softwareName = $params{softwarename};
+	my $softwareVersion = $params{softwareversion};
+	
+	# Registry access for SQL Instances
+    my $machKey = $Registry->Open('LMachine', {Access => KEY_READ}) or $logger->error("Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR");
+    my $sqlinstancesList = $machKey->{"SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL"};
+
+	# List of SQL Instances
+	foreach my $sqlinstance (keys %$sqlinstancesList) {
+		my $sqlinstanceName = $sqlinstance;
+		my $sqlinstanceValue = $sqlinstancesList->{$sqlinstance};
+		# Get version and edition for each instance
+		my ($sqlinstanceEditionValue,$sqlinstanceVersionValue) = _getSqlInstancesVersions(
+			%params,
+			SOFTVERSION	=> $softwareVersion,
+			NAME		=> $sqlinstanceName,
+			VALUE		=> $sqlinstanceValue
+		);
+
+		if ($softwareVersion eq $sqlinstanceVersionValue) {
+			return ($sqlinstanceEditionValue,$sqlinstanceVersionValue);
+		} 
+	}
+}
+
+# SQL Instances versions
+# Return version and edition for each instance
+sub _getSqlInstancesVersions {
+    my (%params) = @_;
+    my $logger = $params{logger};
+
+	my $softwareVersion = $params{SOFTVERSION};	
+	my $sqlinstanceName = $params{NAME};
+	my $sqlinstanceValue = $params{VALUE};
+	
+	my $machKey = $Registry->Open('LMachine', {Access => KEY_READ}) or $logger->error("Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR");
+    my $sqlinstanceVersion = $machKey->{"SOFTWARE/Microsoft/Microsoft SQL Server/" . $sqlinstanceValue . "/Setup"};
+	my $sqlinstanceVersionValue = $sqlinstanceVersion->{'/Version'};
```

Comment:
As $sqlinstanceVersionValue must be defined for later `eq` test, you should return otherwise. You can use the following syntax:
```
    my $sqlinstanceVersionValue = $sqlinstanceVersion->{'/Version'}
        or return;
```
Following previous comment, you should not need to handle a value to return.